### PR TITLE
remove some ResolvedQuery cases

### LINKED ISSUE #14871: Verify that `ResolvedQuery.Empty` is ok where it is used

### PR DESCRIPTION
**Pull request by S11001001**
Fixes #14871.

* exploit non-emptiness to avoid a Monoid law violation
* replace `Empty` by widening the domain of `fromLedgerApi` to ensure that `ResolvedQuery` can also be non-empty
* define the non-empty counterpart of `foldMapM`
* [x] no changelog

### CODE REVIEW

File: ledger-service/fetch-contracts/src/main/scala/fetchcontracts/domain/ContractTypeId.scala [hunk 1]
```diff
@@ -71,17 +71,20 @@ object ResolvedQuery {
...
+        interfaceIds match {
+          case NonEmpty(_) => -\/(CannotQueryBothTemplateIdsAndInterfaceIds)
+          case _ => \/-(ByTemplateIds(templateIds))
+        }
+      case _ =>
+        interfaceIds match {
+          case NonEmpty(Singleton(interfaceId)) => \/-(ByInterfaceId(interfaceId))
+          case NonEmpty(_) => -\/(CannotQueryManyInterfaceIds)
+          case _ => -\/(CannotBeEmpty)
+        }
```
Comment (ray-roestenburg-da) [inline ledger-service/fetch-contracts/src/main/scala/fetchcontracts/domain/ContractTypeId.scala:87]: Can you fix the typo in the error message a little further down:

```
final case object CannotQueryBothTemplateIdsAndInterfaceIds
      extends Unsupported("Cannot resolve any template ID from request")
```

Comment (chunlokling-da) [reply ledger-service/fetch-contracts/src/main/scala/fetchcontracts/domain/ContractTypeId.scala:87]: The original message: 
```
Cannot query both templates ID's and interface ID's
```

Thanks

Comment (S11001001) [reply ledger-service/fetch-contracts/src/main/scala/fetchcontracts/domain/ContractTypeId.scala:87]: ad3e455a371f4393f419dd25f3e87ef8ba36f48d

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/util/Collections.scala [hunk 2]
```diff
@@ -5,15 +5,25 @@ package com.daml.http.util
...
 
-object Collections {
+private[http] object Collections {
 
   implicit final class `cdhuc Nel Ops`[A](private val self: NonEmptyList[A]) extends AnyVal {
     def collect[B](f: A PartialFunction B): Option[NonEmptyList[B]] =
       self.list.collect(f).toNel
   }
 
+  implicit final class `cdhuc Foldable1 Ops`[F[_], A](private val self: F[A]) extends AnyVal {
```
Comment (ray-roestenburg-da) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/util/Collections.scala:18]: cdhuc?

**Review by ray-roestenburg-da — APPROVED** [review]

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/util/Collections.scala (hunk 2 above)
Comment (S11001001) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/util/Collections.scala:18]: Initialism for `com.daml...`

**Pull request merged**